src/game/TileEngine/Render_Dirty.cc

```diff
@@ -355,6 +355,12 @@ void ShutdownBackgroundRects(void)
 		BACKGROUND_SAVE* const b = gBackSaves[i];
 		if (b->fAllocated) FreeBackgroundRectNow(b);
 	}
+	for (auto elem : gBackSaves)
+	{
+		delete elem;
```

Comment:
Why don't we just do `delete b` right after line 356?

Reply:
Because the size of the vector might be larger than `guiNumBackSaves`. `guiNumBackSaves` is basically the number of background rects that the game is currently using, while the vector is resized in increments of 100 [here](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/1dabffb17d4d732a8279597b613fbe6e35b4d54e/src/game/TileEngine/Render_Dirty.cc#L116).

Reply:
I see. In that case, should we move the `FreeBacgroundRectNow(b)` to right before line 360? There is a check for `b->fAllocated` so it should  be safe to run on all of gBackSaves.

Reply:
Sounds good. Done.